examples/play.zig

```diff
@@ -0,0 +1,62 @@
+const std = @import("std");
+const sysaudio = @import("sysaudio");
+const Flac = @import("flac");
+
+var file_decoded: Flac = undefined;
+var player: sysaudio.Player = undefined;
+
+pub fn main() !void {
+    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
+    defer _ = gpa.deinit();
+    const allocator = gpa.allocator();
+
+    var ctx = try sysaudio.Context.init(null, allocator, .{});
+    defer ctx.deinit();
+    try ctx.refresh();
+    const device = ctx.defaultDevice(.playback) orelse return error.NoDevice;
+
+    const audio_file = try std.fs.cwd().openFile("examples/assets/center.flac", .{});
+
+    file_decoded = try Flac.decodeStream(allocator, std.io.StreamSource{ .file = audio_file });
+    defer allocator.free(file_decoded.samples);
+    if (file_decoded.channels > device.channels.len) {
+        return error.InvalidDevice;
+    }
+
+    player = try ctx.createPlayer(device, writeCallback, .{ .format = .i16 });
```

Comment:
very happy with the new example!

on macOS it crashes right now:

```
> thread 1905284 panic: reached unreachable code
/Users/slimsag/.cache/zig/p/12202fe9409b44ea0f1b2212d1ae2259ea8e9ba9d1cccfffb91df83013694bbcb2cc/src/main.zig:233:25: 0x10447ffa3 in write__anon_19200 (example-play)
                else => unreachable,
                        ^
/Users/slimsag/Desktop/hexops/mach-flac/examples/play.zig:58:25: 0x10447728b in writeCallback (example-play)
            player.write(player.channels()[ch], fi, file_decoded.samples[i]);
                        ^
/Users/slimsag/.cache/zig/p/12202fe9409b44ea0f1b2212d1ae2259ea8e9ba9d1cccfffb91df83013694bbcb2cc/src/coreaudio.zig:513:21: 0x1044813ff in renderCallback (example-play)
        self.writeFn(self.user_data, frames);
                    ^
```

Because [there are unimplemented branches here](https://github.com/hexops/mach-sysaudio/blob/b1f45184af53b45a1cb39fb3e29f2e70959da622/src/main.zig#L231) 

Also I am surprised to see we have [a function like this](https://github.com/hexops/mach-sysaudio/blob/b1f45184af53b45a1cb39fb3e29f2e70959da622/src/main.zig#L211) which while operating on a single sample has so much function branching, calculations like `var ptr = channel.ptr + frame * self.writeStep();` and calls to `unsignedToSigned`, `signedToSigned` etc. not marked as inline. That will be very costly / bad for perf.

I will change the type passed so at least the example should run everywhere:

```diff
-player = try ctx.createPlayer(device, writeCallback, .{ .format = .i16 });
+player = try ctx.createPlayer(device, writeCallback, .{ .format = .f32 });
```